### PR TITLE
enable async-profiler by default on x86/x64

### PR DESCRIPTION
**Pull request by richardstartin**
# What Does This Do

Enables async-profiler on any JVM running on x86/x64 or J9

# Motivation

# Additional Notes

Depends on #3992 because this needs to be tested on more J9 and OpenJ9 distributions

### CODE REVIEW

File: dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public final class ProfilingConfig {
   public static final String PROFILING_AUXILIARY_TYPE_DEFAULT = "none";
 
   public static final String PROFILING_ASYNC_ENABLED = "profiling.async.enabled";
-  public static final boolean PROFILING_ASYNC_ENABLED_DEFAULT = false;
+  public static final boolean PROFILING_ASYNC_ENABLED_DEFAULT = System.getProperty("os.arch").contains("x86");;
```
Comment (jbachorik) [inline dd-trace-api/src/main/java/datadog/trace/api/config/ProfilingConfig.java:58]: Would it be possible to use the newly added `Platform.isX86()` method instead?

File: dd-smoke-tests/src/main/groovy/datadog/smoketest/AbstractSmokeTest.groovy [hunk 2]
```diff
@@ -88,7 +89,7 @@ abstract class AbstractSmokeTest extends ProcessManager {
     "-Ddd.profiling.start-delay=${PROFILING_START_DELAY_SECONDS}",
     "-Ddd.profiling.upload.period=${PROFILING_RECORDING_UPLOAD_PERIOD_SECONDS}",
     "-Ddd.profiling.url=${getProfilingUrl()}",
-    "-Ddd.profiling.async.enabled=false",
+    "-Ddd.profiling.async.enabled=${Platform.isX86()}",
```
Comment (jbachorik) [inline dd-smoke-tests/src/main/groovy/datadog/smoketest/AbstractSmokeTest.groovy:92]: IIAM, we can remove this altogether as the default value will be the same (eg. enabled on x86 only)?

**Review by jbachorik — COMMENTED** [review]
Just a few clarification questions ...

File: dd-smoke-tests/src/main/groovy/datadog/smoketest/AbstractSmokeTest.groovy (hunk 2 above)
Comment (richardstartin) [reply dd-smoke-tests/src/main/groovy/datadog/smoketest/AbstractSmokeTest.groovy:92]: I intend to set up tests on aarch64 soon

File: internal-api/src/main/java/datadog/trace/api/Platform.java [hunk 3]
```diff
@@ -230,6 +230,10 @@ public static boolean isJ9() {
         && System.getProperty("java.vm.name").contains("J9");
   }
 
+  public static boolean isX86() {
+    return System.getProperty("os.arch").contains("x86");
```
[the remark this answers is not in the document]
Comment (richardstartin) [reply internal-api/src/main/java/datadog/trace/api/Platform.java:234]: Can it NPE though?

File: dd-smoke-tests/src/main/groovy/datadog/smoketest/AbstractSmokeTest.groovy (hunk 2 above)
Comment (richardstartin) [reply dd-smoke-tests/src/main/groovy/datadog/smoketest/AbstractSmokeTest.groovy:92]: oh yes good point 🤣

**Review by jbachorik — APPROVED** [review]

**Review by jbachorik — APPROVED** [review]

File: dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilingSystem.java [hunk 4]
```diff
@@ -149,6 +149,10 @@ private void startProfilingRecording() {
           uploadPeriod.toMillis(),
           TimeUnit.MILLISECONDS);
       started = true;
+    } catch (UnsupportedEnvironmentException unsupported) {
+      log.warn(
+          "Datadog Profiling was enabled on an unsupported JVM, will not profile application. See {} for more details about supported JVMs.",
+          "https://docs.datadoghq.com/profiler/enabling/java/?tab=commandarguments#requirements");
```
Comment (jbachorik) [inline dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/controller/ProfilingSystem.java:155]: Nice touch!

**Review by jbachorik — APPROVED** [review]

**Pull request merged**